### PR TITLE
[PC-1868] Portenta X8: Release Notes Naming Scheme Update

### PR DESCRIPTION
**Pull request by TaddyHC**
## What This PR Changes
Updates the Release Notes documentation regarding the version naming scheme

## Contribution Guidelines
- [x] I confirm that I have read the [contribution guidelines](https://github.com/arduino/docs-content/tree/main/contribution-templates) and comply with them.

### CODE REVIEW

**Comment by TaddyHC** [comment]
@jcarolinares, we've got approval in the [Jira task](https://arduino.atlassian.net/browse/PC-1868) from @martab1994 . Let me know if you would like suggest something additional, or we could proceed to merge

**Review by jcarolinares — APPROVED** [review]

**Comment by jcarolinares** [comment]
Approved but @TaddyHC DO NOT MERGE

we are waiting for Mattia to check some rendering failures, let's wait!

**Pull request merged**
